components/script/microdata.rs

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::DocumentBinding::DocumentBinding::DocumentMethods;
+use dom::bindings::codegen::Bindings::ElementBinding::ElementBinding::ElementMethods;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::root::DomRoot;
+use dom::document::Document;
+use dom::element::Element;
+use dom::htmlelement::HTMLElement;
+use dom::node::Node;
+use script_traits::Microdata;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+pub fn parse(doc: &Document, node: &Node) -> Option<Microdata> {
+    let serialized_vcard = parse_vcard(doc);
+    let serialized_json = parse_json(node);
+    if !serialized_vcard.is_empty() {
+        return Some(Microdata::VCardData(serialized_vcard.to_owned()));
+    } else if !serialized_vcard.is_empty() {
```

Comment:
Should be `serialized_json` here?

Reply:
Hi, Alex, can you please provide more information about your query? This change addresses the issue https://github.com/servo/servo/issues/18528

Reply:
Lines 20 and 22 are checking the if condition.  It would seem to me that line 22 should be looking at `serialized_json` rather than `serialized_vcard`.